adr/adr.001.folder structure/adr.001.folder structure.md

```diff
@@ -0,0 +1,260 @@
+touch different parts of the code.
+
+## Implementation Details
+
+- **Step 1:** Create a `features/` directory at the root of
+  the project.
+- **Step 2:** Create a `shared/` directory for common
+  components, hooks, utils, and other shared resources.
+- **Step 3:** As new features are developed, implement them
+  within the new structure under `features/`.
+- **Step 4:** When existing code is modified:
+  a. Assess if it belongs to a specific feature.
+  b. If so, move it to the appropriate feature directory.
+  c. If not, consider moving it to the `shared/` directory.
+- **Step 5:** Update import paths for moved files and their
+  dependencies.
+- **Step 6:** Gradually refactor existing features during
+  planned refactoring sprints or as part of regular
+  maintenance.
+- **Step 7:** Regularly review and update documentation to
+  reflect the evolving folder structure.
+- **Step 8:** Conduct team training sessions to ensure all
+  developers understand and follow the new structure.
+
+## File Naming Conventions
+
+We will adopt a consistent naming convention for key files
+within each feature directory. This approach makes it easy
+to search for and identify specific file types, and clearly
+indicates the purpose of each file.
+
+- **Key File Types:**
+  - `[featureName].screen.tsx`: Feature screens
+  - `[featureName].nav.tsx`: Navigation components
+  - `[featureName].store.ts`: Data stores
+  - `[featureName].queries.ts`: API queries - using React Query
+  - `[featureName].utils.ts`: Utility functions
+  - `[featureName].data.ts`: Data models or types
+  - `[featureName].actor.ts`: State machines or actors
+  - `[featureName].client.ts`: Clients that control dependencies
+  - `[featureName].types.ts`: Types for this particular feature
+
+This naming convention offers several benefits:
+1. Easy to search: Use command+P to quickly find all 
+   relevant files for a particular domain.
+2. Clear purpose: File names immediately indicate their 
+   function within the feature.
+3. Consistent organization: Standardizes file naming 
+   across the project.
+
+*Example of the new structure for the Accounts feature:*
+
+```
+features/
+└── accounts/
+    ├── components/
+    │   ├── AccountSettingsButton.tsx
+    │   ├── ExternalWalletPicker.tsx
+    │   ├── EphemeralAccountBanner.tsx
+    │   ├── ProfileSettingsButton.tsx
+    │   └── Logout/
+    │       ├── privy.tsx
+    │       ├── privy.web.tsx
+    │       ├── wallet.tsx
+    │       └── wallet.web.tsx
+    ├── screens/
+    │   ├── Accounts.screen.tsx
+    │   ├── AccountsAndroid.screen.tsx
+    │   └── Onboarding/
+    │       ├── Onboarding.screen.tsx
+    │       ├── Onboarding.web.screen.tsx
+    │       └── components/
+    ├── navigation/
+    │   ├── Accounts.nav.tsx
+    │   ├── AccountsDrawer.nav.tsx
+    │   └── OnboardingNav.nav.tsx
+    ├── hooks/
+    │   ├── useDisconnectActionSheet.ts
+    │   └── usePhotoSelect.ts
+    ├── utils/
+    │   ├── addressBook.ts
```

Comment:
depending on the contents of all of the files in `accounts/utils`, these could all be included in a file `accounts.utils.ts`

Reply:
Personally I like keeping the utils folder for each, as it makes jest mocking much more manageable, just mocking the used function, not the entirety of a huge file

Reply:
> Personally I like keeping the utils folder for each, as it makes jest mocking much more manageable, just mocking the used function, not the entirety of a huge file

Great thing to consider! Ideally `utils` should **not** be mocked or they're doing too much, but that's topic for a different discussion.

Reply:
Where would you put very general input output functions like colors helpers or cleanString or whatever? 😮

Reply:
> Where would you put very general input output functions like colors helpers or cleanString or whatever? 😮

@thierryskoda 

`colors` - I'd imagine this will go in some `features/shared/design-system/theme` 

`helpers` on a per feature basis - like `features/account/account.helpers.ts` or `features/account/utils/account.helper1.ts or account.helper2.ts`

`cleanString` could go in `features/shared/utils/cleanString.ts`